libraries/chain/controller.cpp

```diff
@@ -338,6 +338,8 @@ struct controller_impl {
       set_activation_handler<builtin_protocol_feature_t::get_code_hash>();
       set_activation_handler<builtin_protocol_feature_t::get_block_num>();
       set_activation_handler<builtin_protocol_feature_t::crypto_primitives>();
+      set_activation_handler<builtin_protocol_feature_t::bls_primitives>();
+      bls12_381::init();
```

Comment:
Is this the correct place for the library's `init` function? The call to `init` only needs to happen once when `nodeos` starts. It is a live dispatcher that checks if `adc` and `bmi2` cpu features are available and if so sets the faster asm routines.